pvlib/atmosphere.py

```diff
@@ -423,15 +432,16 @@ def first_solar_spectral_correction(pw, airmass_absolute, module_type=None,
     # Replace Pwat Values below 0.1 cm with 0.1 cm to prevent model from
     # diverging"
 
-    if np.min(pw) < 0.1:
-        pw = np.maximum(pw, 0.1)
-        warn('Exceptionally low Pwat values replaced with 0.1 cm to prevent' +
-             ' model divergence')
+    if np.min(pw) < min_pw:
+        pw = np.maximum(pw, min_pw)
+        warn('Exceptionally low pw values replaced with {0} cm to prevent '
+             'model divergence'.format(min_pw))
 
     # Warn user about Pwat data that is exceptionally high
-    if np.max(pw) > 8:
-        warn('Exceptionally high Pwat values. Check input data:' +
-             ' model may diverge in this range')
+    if np.max(pw) > max_pw:
+        pw[pw > max_pw] *= np.nan
```

Comment:
The slice syntax requires array input. I personally find it extremely useful to be able to use scalar input to understand functions. So, I suggest adding something like `pw = np.atleast1d(pw)`. Why `*=` instead of `=`?

Reply:
Good point.

> Why *= instead of =?
because of a stupid copy and paste